pkgs/build-support/singularity-tools/default.nix

```diff
@@ -108,14 +116,14 @@ rec {
 
             # Build /bin and copy across closure
             mkdir -p bin ./${builtins.storeDir}
-            for f in $(cat $layerClosure) ; do
-              cp -ar $f ./$f
-            done
+            while IFS= read -r f; do
+              cp -r "$f" "./$f"
+            done < "$layerClosure"
 
-            for c in ${toString contents} ; do
-              for f in $c/bin/* ; do
-                if [ ! -e bin/$(basename $f) ] ; then
-                  ln -s $f bin/
+            for c in ${lib.escapeShellArgs contents} ; do
+              for f in "$c"/bin/* ; do
+                if [ ! -e "bin/$(basename "$f")" ] ; then
+                  ln -s "$f" bin/
```

Comment:
May not be in scope for the PR but it's maybe worth linking

- https://github.com/NixOS/nixpkgs/issues/93122
- https://github.com/NixOS/nixpkgs/pull/93082

...from here

Reply:
I agree with using `PATH` instead of `/bin` symlinking. I used it in the definition-based build provided by #224636. Here is our previous discussion on this topic: https://github.com/NixOS/nixpkgs/pull/224636/files#r1160279847

> May not be in scope for the PR

I'm unsure if people (we) have reached a consensus about this change, so I didn't include it in this PR.